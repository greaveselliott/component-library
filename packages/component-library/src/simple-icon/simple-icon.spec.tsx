import React from 'react';

import { shallow } from 'enzyme';
import {
  faExclamationCircle,
  faGlobeAmericas,
  faMousePointer,
} from '@fortawesome/free-solid-svg-icons';
import { IconProp } from '@fortawesome/fontawesome-svg-core';

import SimpleIcon, {
  SimpleIconColours,
  SimpleIconVariant,
} from './simple-icon.component';

jest.mock('@fortawesome/react-fontawesome', () => ({
  FontAwesomeIcon: () => null,
}));

jest.mock('@fortawesome/free-solid-svg-icons', () => ({
  faExclamationCircle: () => null,
  faGlobeAmericas: () => null,
  faMousePointer: () => null,
}));

export default { title: 'Simple Icon' };

const iconColors: SimpleIconColours[] = [
  'primary-400',
  'shade-400',
  'accent-200',
  'accent-300',
  'accent-400',
];

const iconVariants: SimpleIconVariant[] = ['small', 'large'];

const iconSet: IconProp[] = [
  faExclamationCircle,
  faGlobeAmericas,
  faMousePointer,
];

iconVariants.forEach((variant) =>
  iconColors.forEach((color) =>
    iconSet.forEach((icon) => {
      test(`Render the ${variant} ${color} ${icon} icon`, () => {
        const SUT = shallow(
          <SimpleIcon variant={variant} color={color} icon={icon} />,
        );
        expect(SUT).toMatchSnapshot();
      });
    }),
  ),
);
