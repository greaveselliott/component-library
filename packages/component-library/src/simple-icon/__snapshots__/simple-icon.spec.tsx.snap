// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the large accent-200 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-200"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-200 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-200"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-200 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-200"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-300 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-300"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-300 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-300"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-300 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-300"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-400 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-400 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large accent-400 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-accent-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large primary-400 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-primary-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large primary-400 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-primary-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large primary-400 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-primary-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large shade-400 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-shade-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large shade-400 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-shade-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the large shade-400 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--large simple-icon--color-shade-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-200 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-200"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-200 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-200"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-200 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-200"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-300 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-300"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-300 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-300"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-300 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-300"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-400 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-400 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small accent-400 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-accent-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small primary-400 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-primary-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small primary-400 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-primary-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small primary-400 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-primary-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small shade-400 function () { return null; } icon 1`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-shade-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small shade-400 function () { return null; } icon 2`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-shade-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;

exports[`Render the small shade-400 function () { return null; } icon 3`] = `
<div
  className="simple-icon simple-icon--small simple-icon--color-shade-400"
>
  <FontAwesomeIcon
    icon={[Function]}
  />
</div>
`;
